go-controller/pkg/clustermanager/network_cluster_controller.go

```diff
@@ -327,8 +402,10 @@ func (h *networkClusterControllerEventHandler) AddResource(obj interface{}, from
 		if err = h.ncc.nodeAllocator.HandleAddUpdateNodeEvent(node); err != nil {
 			klog.Infof("Node add failed for %s, will try again later: %v",
 				node.Name, err)
+			h.ncc.updateNetworkStatus(node.Name, fmt.Sprintf("Node add failed: %v", err))
```

Comment:
The idea we would just do this when we are done with retries or at least on a long backoff.

I don't know if it can happen in this specific case but in general it might be expected that we fail until something else happens in the system and we might end up reporting spurious errors that while fast enough to go unnoticed by humans, might trigger machines.

Reply:
yeah I like that idea, but I think it only works for controllers setting their retry queues themselves, and for thos eusing obj_retry it is difficult to get that info.
On the bright side: new events will only be recorded if an error message changes, and retry_obj only retries every 30 sec, so at least we should not overload api server with events.

Reply:
> for thos eusing obj_retry it is difficult to get that info

So the retry framework does provide `RecordErrorEvent` callback that can be implemented. If that doesn't suit our purpose we could just add a different callback that does.

> retry_obj only retries every 30 sec

we should be prepared for this to changed in the future

You can convince me though to maybe tackle this one later.

Reply:
I guess it is not only about the retry period, but also about just everything potentially failing on the first attempt.

Reply:
I think you may want to skip reporting an error immediately for 2 reasons:
1. you expect next retry to succeed or the error message to change
2. you are trying to reduce the error reporting load on the kapi server.
Otherwise I think reporting an error immediately is the best choice. 
So in this case, waiting for 30 seconds to see the error reported may be too much, e.g. if the error is something easy to fix, like too small node subnet in the UDN spec. Also I don't expect this specific case to overload the apiserver. BUT if we ever change the retry logic, this definitely should be reconsidered.
But if we have some exponential backoff, and will have 5 retries in a seconds or something around that, then it makes sense to report error on like 5th retry, since it follows both reasons. wdyt?

Reply:
If you think it has a high chance to fail on the first try but succeed on the second, I could add a logic to report error starting with the second retry only.
But here I am also thinking about the condition itself: if I don't report an error, and I can't report success, should I have that condition as Unknown until next retry? Kinda knowing at the same time there is a failure, and if a pod in that namespace failed to start it is because of that failure. That could work for fast retries, but not sure if 30s is a good time to keep unknown status

Reply:
OK so we agree that we need to find a deadline when it is reasonable to send the status report. And for our retry framework to send that status report once on the retry closest to that deadline or soemthing similar.

Let's say we think today that deadline is 30s. Oh! It matches our retry framework deadline! What a coincidence! 
 :P 

That would mean we would report once on the second attempt as you suggest.

However I don't know if I would like to hack this now or follow up with a proper implementation. Will let you decide.